### PR TITLE
add package level documentation

### PR DESCRIPTION
**Pull request by billwert**

### CODE REVIEW

**Review by mssfang — APPROVED** [review]

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/package-info.java [hunk 1]
```diff
@@ -1,5 +1,77 @@
...
+ *
+ * <p><strong>Key Concepts:</strong></p>
+ * <ul>
+ *     <li><strong>Events</strong> - Information about what happened.</li>
+ *     <li><strong>Event Source</strong> - where the event took place.</li>
+ *     <li><strong>Topics</strong> - the endpoint where events are published to.</li>
+ *     <li><strong>Event Handlers</strong> - the endpoint that handles the events.</li>
+ *     <li><strong>Event Subscriptions</strong> - the endpoint or built-in mechanism for routing events.</li>
+ * </ul>
+ * <p>For more information see the <a href="https://learn.microsoft.com/en-us/azure/event-grid/concepts">concepts overview.</a></p>
```
Comment (g2vinay) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/package-info.java:17]: Its missing a Getting Started section with Authentication info. 

Metrics Advisor's Getting Started section here can be used as an example: 
https://azuresdkdocs.blob.core.windows.net/$web/java/azure-ai-metricsadvisor/1.1.21/com/azure/ai/metricsadvisor/package-summary.html

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/package-info.java [hunk 2]
```diff
@@ -1,5 +1,77 @@
...
+ * <h2>Send an EventGridEvent</h2>
```
Comment (g2vinay) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/package-info.java:18]: add new line before and after, makes it easy to read in IDE.

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/package-info.java [hunk 3]
```diff
@@ -1,5 +1,77 @@
...
+ *
+ * &#47;&#47; Send a list of CloudEvents to the EventGrid service altogether.
+ * &#47;&#47; This has better performance than sending one by one.
+ * cloudEventPublisherClient.sendEvents&#40;Arrays.asList&#40;
+ *     cloudEventDataObject
+ *     &#47;&#47; add more CloudEvents objects
+ * &#41;&#41;.block&#40;&#41;;
+ * </pre>
+ * <!-- end com.azure.messaging.eventgrid.EventGridPublisherAsyncClient#SendCloudEvent -->
+ */
```
Comment (g2vinay) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/package-info.java:76]: add see tags to clients, builders and any other important class.

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/package-info.java [hunk 4]
```diff
@@ -1,5 +1,77 @@
...
+ * <ul>
+ *     <li><strong>Events</strong> - Information about what happened.</li>
+ *     <li><strong>Event Source</strong> - where the event took place.</li>
+ *     <li><strong>Topics</strong> - the endpoint where events are published to.</li>
+ *     <li><strong>Event Handlers</strong> - the endpoint that handles the events.</li>
+ *     <li><strong>Event Subscriptions</strong> - the endpoint or built-in mechanism for routing events.</li>
+ * </ul>
+ * <p>For more information see the <a href="https://learn.microsoft.com/en-us/azure/event-grid/concepts">concepts overview.</a></p>
+ * <h2>Send an EventGridEvent</h2>
+ * <p>In order to interact with the Azure Event Grid service, you will need to create an instance of the {@link com.azure.messaging.eventgrid.EventGridPublisherClient} class:</p>
```
Comment (g2vinay) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/package-info.java:19]: add new line after,  makes it easy to read in IDE.

**Pull request merged**